### PR TITLE
Upgrade Go, Infra components

### PR DESCRIPTION
**Pull request by krithika369**
The main changes arise from the PR https://github.com/gojek/turing/pull/183 which upgraded the infrastructure dependencies of Turing and updated the codebase and CI scripts accordingly.

As noted in README.md, the following is the summary of the changes:
- Golang: 0.14 -> 0.18
- Kubectl: v0.19 -> v1.22 (for the CI steps)
- Knative: v0.18.3 -> v1.0.1
- Knative Istio: v0.18.1 -> v1.0.0
- Istio: 1.9.9 -> 1.12.5

### CODE REVIEW

**Comment by krithika369** [comment]
Most of the changes are in `go.sum`.

File: api/e2e/test/00_all_e2e_test.go [hunk 1]
```diff
@@ -1,5 +1,4 @@
 //go:build e2e
-// +build e2e
```
Comment (leonlnj) [inline api/e2e/test/00_all_e2e_test.go:2]: TIL

https://stackoverflow.com/questions/68360688/whats-the-difference-between-gobuild-and-build-directives

File: infra/docker-compose/dev/istio/minimal-operator.yaml [hunk 2]
```diff
@@ -5,38 +5,22 @@ spec:
...
       enabled: true
-    - name: cluster-local-gateway
-      enabled: true
-      label:
-        istio: cluster-local-gateway
-        app: cluster-local-gateway
       k8s:
         service:
-          type: ClusterIP
+          type: LoadBalancer
```
Comment (leonlnj) [inline infra/docker-compose/dev/istio/minimal-operator.yaml:17]: whats the motivation for the change from `ClusterIp` to `LoadBalancer`?

**Review by leonlnj — APPROVED** [review]

File: infra/docker-compose/dev/istio/minimal-operator.yaml (hunk 2 above)
Comment (krithika369) [reply infra/docker-compose/dev/istio/minimal-operator.yaml:17]: Good question. AFAIK, Ingress gateways are typically set to `LoadBalancer` to be accessible outside the cluster. I think `ClusterIP` can also be used with k3d in which case the Istio ingress gateway needs to be exposed differently (see `.github/actions/run-cluster-init/action.yaml`).

File: api/turing/cluster/controller.go [hunk 3]
```diff
@@ -385,117 +393,125 @@ func (c *controller) DeployKubernetesService(
...
 
 // DeleteIstioVirtualService deletes an istio virtual service.
-func (c *controller) DeleteIstioVirtualService(svcName string, namespace string, timeout time.Duration) error {
+func (c *controller) DeleteIstioVirtualService(
+	ctx context.Context,
+	svcName string,
+	namespace string,
+	timeout time.Duration,
+) error {
+	// It is safe to upgrade this to v1beta1 as it can delete v1alpha3 versions of VirtualService as well.
```
[the remark this answers is not in the document]
Comment (krithika369) [reply api/turing/cluster/controller.go:464]: Thanks for catching. That comment was indeed better for the original MR it was introduced in and carries little meaning now. I removed it.

**Pull request merged**